linux/scripts/upload-to-debian.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+# Upload the latest release to mentors.debian.net. Call from stable-* branch.
+# Usage: upload-to-debian.sh -k <Debian signing key>
+# Requires an entry in ~/.dput.cf:
+# ```
+# [mentors]
+# fqdn = mentors.debian.net
+# incoming = /upload
+# method = https
+# allow_unsigned_uploads = 0
+# progress_indicator = 2
+# allowed_distributions = .*
+# ```
+
+set -e
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/../../resources/build/build-utils.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+PROGRAM_NAME="$(basename "$0")"
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m' # No Color
+
+error()
+{
+	echo -e "${RED}$PROGRAM_NAME: $1${NC}" >&2
+}
+
+log()
+{
+	echo -e "${GREEN}$PROGRAM_NAME: $1${NC}"
+}
+
+usage()
+{
+    echo ""
+    echo "Usage:"
+    echo "${0} -k <Debian signing key> [-n] [--push]"
+    echo ""
+    echo "Create source package for latest stable release and upload to Debian,"
+    echo "and update (and optionally push) changelog file."
+    echo ""
+    echo "    -k <key>    The PGP key used to sign the source package"
+    echo "    -n          Simulate upload"
+    echo "    --push      Push changelog changes to GitHub"
+    echo "    --help      Display usage help"
+}
+
+while (( $# )); do
+    case $1 in
+        -k) shift; debkeyid=$1;;
+        -n) NOOP=: ;;
+        --help) usage ; exit 0 ;;
+        --push) PUSH=1 ;;
+        *) error "Error: Unexpected argument \"$1\". Exiting." ; exit 1 ;;
+    esac
+    shift || (error "Error: The last argument is missing a value. Exiting."; false) || exit 2
+done
+
+if [ -z "$debkeyid" ]; then
+    usage
+    exit 3
+fi
+
+if ! git diff --quiet; then
+    error "You have changed files in your git working directory. Exiting."
+    exit 4
+fi
```

Comment:
I wonder if we should be using ``if [[ `git status --porcelain` ]]; then`` instead?

https://stackoverflow.com/a/25149786/1836776

So many ways to skin a cat but `git diff` will miss untracked files, won't it?

Reply:
I don't think we care about untracked files in this script. We only modify `changelog`, which already exists. There could be some untracked files which would be overwritten by switching branches, but then `git checkout` will fail. And there might be untracked files which won't be overwritten, but might be in the workspace for a reason. 

We could use `if [[ `git status --porcelain --untracked-files=no` ]]; then` to ignore untracked files, but than that's not much different from what we have, so I'll leave that.